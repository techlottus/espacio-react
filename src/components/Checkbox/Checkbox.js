import React, { useEffect } from "react";



export const Checkbox = React.memo(({ data, onCheck }) => {
  const checkboxRef = React.createRef();

  useEffect(() => {
    checkboxRef.current.data = {
      label: data.label || "",
      disabled: data.disabled,
      selected: data.selected,
      name: data.name || "",
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    checkboxRef.current.addEventListener("onCheck", onCheck);
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-checkbox ref={checkboxRef}></lottus-checkbox>
    </>
  );
});
