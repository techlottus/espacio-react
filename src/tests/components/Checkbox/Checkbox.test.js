import React from "react";
import { mount } from "enzyme";
import { Checkbox } from "../../../components/Checkbox/Checkbox";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState = {
  data: {
    label: null,
    disabled: null,
    selected: null,
    name: null,
  },
};

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Checkbox data={initState.data} />
    </Provider>
  );
};

describe("Test <Checkbox />", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });
});
