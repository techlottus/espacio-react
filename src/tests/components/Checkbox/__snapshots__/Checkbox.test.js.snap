// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Checkbox /> should init component 1`] = `<lottus-checkbox />`;
